src/coralogix.js

```diff
@@ -147,13 +147,21 @@ class CoralogixLogger extends FormattedLoggerBase {
         subsystemName: subsystem,
         computerName: host,
         logEntries: [{
-          timestamp: Number(BigDate.preciseTime(fields.timestamp)),
+          timestamp: CoralogixLogger._preciseTime(fields.timestamp),
```

Comment:
Can you provide details on what causes the error to be thrown?

Reply:
I'm afraid not: the value for `fields.timestamp` is never displayed.

Reply:
Personally I believe that it is only really possible to develop software and fix bugs effectively long term if we have a complete picture of what is happening. Your solution seems to imply that the timestamp provided is broken but that this error can be safely ignored and that the timestamp is from right about now so it is relatively safe to substitute the current time (you might want to use a high precision time stamp though, that's what big date is there for). Where these assumptions verified?

I think, that the behavior of throwing on incorrect timestamps is probably correct, because indicates some other underlying error (your test seems to imply this; surely something is amiss if the timestamp is 'boo'). I would approach this by improving the diagnostics generated by BigData and then using the improved diagnostics to fix the underlying bug.

Doing this sort of investigation is a lot more time consuming and a fix like the one you are proposing is likely to work good enough for some time, while on the other hand it might hamper our ability to debug problems in the future and could threaten the long term stability of the code base. I have used the quick fix route a lot in harmonicobserver, and appropriately so I think.

Code wise this PR is fine. I'll leave the choice of which trade off is appropriate here up to you.

Reply:
Finding the actual cause of an error is also my preference, but in the case of `helix-log`, being a library used in various Helix services, it is quite hard to find the root cause of the error. One would need to create a new version of the library with diagnostic output, and replace it in all clients of that library.

I agree, though, that it should not simply be ignored, particularly because the error happens quite frequently - once per minute on average - and I'll provide a fix where the offending timestamp is logged as well.